### PR TITLE
Convert outstanding commands to cobra

### PR DESCRIPTION
**Pull request by stephenfin**
**What this PR does / why we need it**:

This PR builds on the work from #2325 and migrates the two remaining commands - namely `client-keystone-auth` and `k8s-keystone-auth` - to Cobra.

Along the way, we do some additional cleanup, removing the unnecessary registration of klog options first noted in [#2325](https://github.com/kubernetes/cloud-provider-openstack/pull/2325#issuecomment-1710381237), fixing the binary name show in the output of `client-keystone-auth`, and removing erroneous output from `magnum-auto-healer --help` caused by our prior use of `init` functions.

The commit messages contain a lot of context and I would recommend looking at those separately.

- `magnum-auto-healer: Stop registering klog options`
- `client-keystone-auth: Stop registering klog options`
- `cinder-csi-plugin: Use binary name in help page`
- `client-keystone-auth: Migrate to cobra`
- `k8s-keystone-auth: Migrate to cobra`
- `occm: Remove unnecessary flag handling code`
- `occm: Register additional options correctly`
- `magnum-auto-healer: Remove use of 'init' methods`

**Which issue this PR fixes(if applicable)**:
N/A

**Special notes for reviewers**:
N/A

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dulek** [comment]
Looks good, check error is valid though.

**Comment by kayrus** [comment]
@stephenfin can you fix the https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2384/pull-cloud-provider-openstack-check/1709258266699108352 issue?

**Comment by stephenfin** [comment]
> @stephenfin can you fix the [prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2384/pull-cloud-provider-openstack-check/1709258266699108352](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2384/pull-cloud-provider-openstack-check/1709258266699108352) issue?

Done

File: cmd/k8s-keystone-auth/main.go [hunk 1]
```diff
@@ -15,47 +15,45 @@ limitations under the License.
...
+
+		},
+		Version: version.Version,
+	}
 
-	logs.AddFlags(pflag.CommandLine)
 	keystone.AddExtraFlags(pflag.CommandLine)
 
-	logs.InitLogs()
-	defer logs.FlushLogs()
```
Comment (kayrus) [inline cmd/k8s-keystone-auth/main.go:38]: I wonder whether removing this may result in missing logs or error messages?

Comment (stephenfin) [reply cmd/k8s-keystone-auth/main.go:38]: This should not be necessary as `cli.Run` takes care of this for us [1][2][3]

[1] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L45
[2] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L88
[3] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L90

File: cmd/k8s-keystone-auth/main.go [hunk 2]
```diff
@@ -15,47 +15,45 @@ limitations under the License.
...
+
 func main() {
-	var showVersion bool
-	pflag.BoolVar(&showVersion, "version", false, "Show current version and exit")
+	cmd := &cobra.Command{
+		Use:   "k8s-keystone-auth",
+		Short: "Keystone authentication webhook plugin for Kubernetes",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := config.ValidateFlags(); err != nil {
+				klog.Errorf("%v", err)
```
[the remark this answers is not in the document]
Comment (stephenfin) [reply cmd/k8s-keystone-auth/main.go:37]: Probably, yes. I've intentionally kept this as simple as possible thought to expedite review process. I can stick an additional commit into the series or address this in another follow-up PR, whatever you'd prefer :slightly_smiling_face:

**Review by stephenfin — COMMENTED** [review]
Let me know what you'd like me to do RE: that potential follow-up change.

**Comment by stephenfin** [comment]
> See my new comments. In addition I'd be very careful with these changes. According to my experience, flags behavior could be a bit fragile and break in certain conditions. Have you double-checked whether all binaries still correctly parse the flags?

I have checked that the `--help` output for each command shows the same flags using a variation of the hacky af script I noted in https://github.com/kubernetes/cloud-provider-openstack/pull/2325. I think we're good here, but a second pair of eyes would not be a disadvantage!

File: cmd/k8s-keystone-auth/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kayrus) [reply cmd/k8s-keystone-auth/main.go:37]: From my point of view it's better to adjust this in one PR. It'll be hard to track similar changes across multiple commits. In addition could you please align the `RunE`/`PreRunE` code with other CPO components? Thanks, in advance.

[the remark this answers is not in the document]
Comment (stephenfin) [reply cmd/k8s-keystone-auth/main.go:37]: I think this needs a little more discussion. The failure modes for different components is very different. There's only one user of `RunE` currently (the little-known `barbican-kms-plugin`) and the rest are using `Run`. There are no users of `PreRunE`. In addition, some commands are using logging (`klog`) for error messages (like this command) while others are printing straight out to `stderr` (like `client-keystone-auth`). If we want to "standardise" all components, we probably want to decide an answer to the following:

- When do we dump whole error messages on failure and when do we wrap with our own errors?
- Do we print errors to stderr or use the logging infrastructure?
- Do we want to place all logic inside the `Run`/`RunE` implementation or in an external `handle` method as we currently do in some places?
- Do we want to use `RunE` consistently, and if so what do we do about processes that never return (as in the `cinder-csi-plugin`, which ultimately creates a `grpc.Server` instance)?
- Do we want to move logic for the current "stub" implementations like e.g. `magnum-auto-healer` from `pkg` to `cmd` (specifically `pkg/autohealing/cmd/root.go` to `cmd/magnum-auto-healer/main.go`)?

I'm happy to address these, but I don't want to do this here as it has a possibility of blowing up very quickly (both due to lines of code and picking colours of bikesheds 😇). Let's do it separately?

**Comment by stephenfin** [comment]
@kayrus Are you okay with my suggestion that we punt the outstanding work to a follow-up and merge this as-is?

**Comment by dulek** [comment]
/lgtm

Let's finally get this through ,we'll fix issues along the way.

**Pull request merged**